lib/gateway/Shard.js

```diff
@@ -1129,11 +1131,17 @@ class Shard extends EventEmitter {
                 if(!(this.client.shards.options.intents & Constants.Intents.guildPresences) && packet.d.user.username !== undefined) {
                     let user = this.client.users.get(packet.d.user.id);
                     let oldUser = null;
-                    if(user && (user.username !== packet.d.user.username || user.discriminator !== packet.d.user.discriminator || user.avatar !== packet.d.user.avatar || user.globalName !== packet.d.user.global_name)) {
+                    if(user && (user.username !== packet.d.user.username
+                        || user.discriminator !== packet.d.user.discriminator
+                        || user.avatar !== packet.d.user.avatar
+                        || user.globalName !== packet.d.user.global_name
+                        || user.avatarDecorationData?.asset !== packet.d.user.avatar_decoration_data?.asset
+                    )) {
                         oldUser = {
                             username: user.username,
                             discriminator: user.discriminator,
                             avatar: user.avatar,
+                            avatar_decoration_data: user.avatarDecorationData,
```

Comment:
Use camel-case to be consistent with the rest of the structure
```suggestion
                            avatarDecorationData: user.avatarDecorationData,
```